src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Primitives
+{
+    public struct StringSegment : IEquatable<StringSegment>
+    {
+        private readonly string _buffer;
+        private readonly int _offset;
+        private readonly int _count;
```

Comment:
Could you get rid of the backing fields as part of this move? Get-only auto-properties seem to work fine in `struct`s now.